src/components/GlobalPreferences/Notifications/NotificationsLogin.js

```diff
@@ -1,72 +1,100 @@
+  Box,
+  Button,
+  GU,
+  TextInput,
+  Info,
+  IconMail,
+  Text,
+  useTheme,
+} from '@aragon/ui'
 import { login } from './notification-service-api'
-import { getEthNetworkType } from '../../../local-settings'
+import { network } from '../../../environment'
+import { validateEmail } from '../../../utils'
 import notificationSvg from './notifications.svg'
 
-export default function NotificationsLogin({
-  dao,
-  authState,
-  email,
-  onEmailChange,
-}) {
+export default function NotificationsLogin({ dao, authState, onEmailChange }) {
   const [inputEmail, setInputEmail] = useState('')
+  const [emailInvalid, setEmailInvalid] = useState(false)
   const [apiError, setApiError] = useState(null)
-  const network = getEthNetworkType()
+  const ethNetwork = network.type === 'main' ? 'mainnet' : 'rinkeby'
 
   const handleEmailChange = e => {
+    setEmailInvalid(!validateEmail(e.target.value))
+
     setInputEmail(e.target.value)
   }
 
   const handleLogin = async e => {
+    e && e.preventDefault()
     try {
-      await login({ email: inputEmail, dao, network })
+      await login({ email: inputEmail, dao, network: ethNetwork })
       onEmailChange(inputEmail)
     } catch (e) {
       setApiError(e.message)
       console.error('Failed to login', e)
     }
   }
 
+  const theme = useTheme()
   return (
     <Box heading="Email notifications">
       <NotificationImage />
-      {apiError && <Info mode="error">Error logging in.</Info>}
-      <Label>Email address</Label>
-      <div
-        css={`
-          display: grid;
-          align-items: center;
-          grid-template-columns: 1fr auto;
-          grid-gap: ${2 * GU}px;
-          margin-bottom: ${2 * GU}px;
-        `}
-      >
-        <TextInput
-          type="email"
-          placeholder="you@example.com"
-          wide
-          value={inputEmail}
-          onChange={handleEmailChange}
-        />
-        <Button
+      <form onSubmit={handleLogin}>
+        <div
           css={`
-            width: ${16 * GU}px;
+            display: grid;
+            align-items: end;
+            grid-template-columns: 1fr auto;
+            grid-gap: ${GU}px ${2 * GU}px;
+            margin-bottom: ${2 * GU}px;
           `}
-          onClick={handleLogin}
         >
-          <IconMail /> Sign in
-        </Button>
-      </div>
+          <label
+            css={`
+              color: ${theme.surfaceContentSecondary};
+              display: block;
+            `}
+          >
+            Email address
+            <TextInput
+              type="email"
+              placeholder="you@example.com"
+              wide
+              value={inputEmail}
+              onChange={handleEmailChange}
+            />
+          </label>
+
+          <Button
+            disabled={emailInvalid || inputEmail.length === 0}
+            css={`
+              width: ${16 * GU}px;
+            `}
+            onClick={handleLogin}
+          >
+            <IconMail /> Sign in
```

Comment:
We should use `Button`'s new `icon` and `label` props to let it control the internal spacing:

```js
<Button
   ...
   icon={<IconMail />
   label="Sign in"
/>
```